cards/src/main/resources/cards/custom/group6/witch_doctor/secrets/secret_greater_good.json

```diff
@@ -0,0 +1,34 @@
+{
+  "name": "Greater Good",
+  "baseManaCost": 2,
+  "type": "SPELL",
+  "heroClass": "ROSE",
+  "rarity": "RARE",
+  "description": "Secret: When your opponent casts a spell, discard a random card from your hand to Counter it.",
+  "secret": {
+    "class": "SpellCastedTrigger",
+    "fireCondition": {
+      "class": "HoldsCardCondition",

```

Comment:
should be a `CardCountCondition` and needs a test, make sure the appropriate `"targetPlayer"` is set